### ISSUE TITLE
More descriptive title on More Information link

### ISSUE THREAD

**Issue by tomheath** [theodi/member-directory#118]
The "More information" link next to each member entry in the directory should have a more descriptive title (e.g. "More information on ACME Corp"), because accessibility.
